### DISCUSSION TITLE
I need some advice on how to update a 4 year old app hosted in vercel

### DISCUSSION THREAD

**Discussion by johhansantana** [vercel/community#1561] [Help] [answered]
### Summary

Hello. I'm not sure if this is the right place to ask this, if not my apologies.

I'm in the task to update a small but 4 year old web app hosted in (back then now.sh) vercel and I've encountered an error when just trying to upload it like before (I haven't used vercel in a long while).

The nextjs app is at 
```"next": "^8.0.4"```

And it has a file called `now.sh` with the following

```json
{
  "version": 2,
  "name": "foobar",
  "alias": "foobar.com",
  "builds": [
    {
      "src": "next.config.js",
      "use": "@now/next@0.0.79",
      "config": { "maxLambdaSize": "35mb" }
    },
    { "src": "server/transpiled/*.js", "use": "@now/node-server" },
    { "src": "static/**/*", "use": "@now/static" }
  ],
  "routes": [
    { "src": "/service-worker.js", "dest": "/static/service-worker.js" },
    { "src": "/sw.js", "dest": "/static/service-worker.js" },
    { "src": "/v1/send-email", "dest": "/server/transpiled/email.js" },
    { "src": "/w/(.*)", "dest": "/work?slug=$1" }
  ],
  "env": {
    "SOME_SECRET": "@secret-key"
  }
}
```

If I try to deploy using this command `vercel` I get an error:
```
normalizing package.json
Error: Unexpected end of JSON input
```

But I've validated that my package.json as well. So I'm assuming there's a new way of doing things and I was wondering if someone could point me in the right direction here.

Thank you

**Comment by amyegan** [comment, marked as answer]
Thanks for reaching out, Johhan. You'll need to convert your `now.sh` into a [`vercel.json` config](https://vercel.com/docs/concepts/projects/project-configuration#) and upgrade any `@now/*` dependencies to their `@vercel` equivalents. There have been a lot of improvements with more recent versions of Next.js, so it's a good idea to consider [upgrading your Next.js version](https://nextjs.org/docs/upgrading) as well.
